ENGAGEHF/Medications/MedicationsList+ViewModel.swift

```diff
@@ -0,0 +1,35 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+
+extension MedicationsList {
+    class ViewModel {
+        let containsRecommendations: Bool
+        
+        let currentlyTakenMedications: [MedicationDetails]
+        let notCurrentlyTakenMedications: [MedicationDetails]
+        
+        
+        init(_ medications: [MedicationDetails]) {
+            // A medication is marked as currently being taken if it contains a non-zero dosage schedule.
+            self.currentlyTakenMedications = {
+                medications.filter { !$0.dosageInformation.currentDailyIntake.isZero }
+            }()
```

Comment:
Remove the closure and just compute the result directly

```suggestion
        
        init(_ medications: [MedicationDetails]) {
            // A medication is marked as currently being taken if it contains a non-zero dosage schedule.
            self.currentlyTakenMedications = medications.filter { !$0.dosageInformation.currentDailyIntake.isZero }
```

also the same thing for the following lines